resources/lib/youtube_plugin/kodion/impl/xbmc/xbmc_items.py

```diff
@@ -47,31 +64,53 @@ def to_play_item(context, play_item):
         else:
             list_item.setArt({'icon': thumb, 'thumb': thumb, 'fanart': fanart})
 
-    if not play_item.use_dash() and not settings.is_support_alternative_player_enabled() and \
-            play_item.get_headers() and play_item.get_uri().startswith('http') and major_version < 20:
-        play_item.set_uri('|'.join([play_item.get_uri(), play_item.get_headers()]))
-
     if settings.is_support_alternative_player_enabled() and \
             settings.alternative_player_web_urls() and \
             not play_item.get_license_key():
         play_item.set_uri('https://www.youtube.com/watch?v={video_id}'.format(video_id=play_item.video_id))
 
-    if play_item.use_dash() and context.addon_enabled('inputstream.adaptive'):
-        inputstream_property = 'inputstream'
-        if major_version < 19:
-            inputstream_property += 'addon'
-       
+    ia_enabled = context.addon_enabled('inputstream.adaptive')
+
+    if ia_enabled and play_item.use_mpd_video() and not play_item.live:
         list_item.setContentLookup(False)
         list_item.setMimeType('application/xml+dash')
-        list_item.setProperty(inputstream_property, 'inputstream.adaptive')
+        list_item.setProperty('inputstream', 'inputstream.adaptive')
```

Comment:
It was removed in line 59-63, but the property name needs to be 'inputstreamaddon' for Kodi 18.